### DISCUSSION TITLE
Support for Markdown-IT plugins

### DISCUSSION THREAD

**Discussion by maldworth** [structurizr/ui#76] [Ideas]
Just wanted to gauge interest in allowing plugins for markdown-it.

For example, I'd like to use some of the syntax for a flow chart, using mermaid. So have a mermaid code fence, and then mermaid syntax.

```code
graph TD;
    A-->B;
    A-->C;
    B-->D;
    C-->D;
```

Is this something you'd consider adding, or accepting a contribution?

**Comment by maldworth** [comment]
I'd like to work on contributing this, but don't want to spend effort if it will never be accepted as a PR. Any update?

**Comment by simonbrowndotje** [comment]
I'm not keen on adding markdown-it plugins ... with Mermaid specifically (1) there are several competing plugins and (2) the Mermaid server needs to be configurable for privacy reasons. Instead I'd recommend pre-processing Mermaid blocks in Markdown files via a DSL plugin, an example of which can be found at https://github.com/structurizr/examples/tree/main/dsl/plantuml-and-mermaid/plugin/src/main/java/mermaid

**Comment by maldworth** [comment]
Oh wow awesome, I had no clue this example existed! I searched the docs, but there was only the mermaid exporter.

So my only question about the plugin.jar file that I will build against the maven dependency of structurizr-dsl `implementation 'com.structurizr:structurizr-dsl:1.30.2'` [here](https://github.com/structurizr/examples/blob/main/dsl/plantuml-and-mermaid/plugin/build.gradle). Every time that we update and use a newer version of structurizr-lite, do I have to make sure to re-build the plugin against the same version?

For example https://central.sonatype.com/artifact/com.structurizr/structurizr-dsl/1.30.2 is quite old, from 2023. So I imagine that plugin.jar I generate from that would not work with the latest structurizr-lite docker image.

How forgiving is it with dependency 3.0.^? or 3.^?

**Reply by simonbrowndotje** [reply]
> Every time that we update and use a newer version of structurizr-lite, do I have to make sure to re-build the plugin against the same version?

Yes, you should ideally recompile your plugin each time you update Lite and it uses a new version of the `structurizr-dsl` library.
